### ISSUE TITLE
Listing service groups returns only a single service group

### ISSUE THREAD

**Issue by dberenguer** [telefonicaid/iotagent-ul#391]
This documentation:
https://fiware-tutorials.readthedocs.io/en/latest/iot-agent/index.html#service-group-crud-actions

Describes the API call to get the list of service groups available under a service path:

REQUEST:
```
curl -X GET \
  'http://localhost:4041/iot/services' \
  -H 'fiware-service: openiot' \
  -H 'fiware-servicepath: /'
```

RESONSE:
```
{
    "_id": "5b07b2c3d7eec57836ecfed4",
    "subservice": "/",
    "service": "openiot",
    "apikey": "4jggokgpepnvsb2uv4s40d59ov",
    "resource": "/iot/d",
    "attributes": [],
    "lazy": [],
    "commands": [],
    "entity_type": "Thing",
    "internal_attributes": [],
    "static_attributes": []
}
```

However both the documentation and our tests result in the UL agent returning a single object (service group) instead of a list.

Is this the normal behaviour? How should we proceed in order to get the list of service groups?

**Comment by dberenguer** [comment]
**Update:**

Passing IOTA_REGISTRY_TYPE to mongodb fixes this problem. This is a bug in our opinion since the behavior of the API should be the same despite any config value. The docs also point to what we think is the incorrect behavior.

```
IOTA_REGISTRY_TYPE - Whether to hold IoT device info in memory or in a database
```

**Comment by fgalan** [comment]
I agree the API of the IOTA should be the same no matter the configured registry type (either memory or mongo). Thus, this is a bug (although a little one... as it works with mongo as registry, which is the one used in real-world scenarios).

Thanks for the feedback!

**Comment by fgalan** [comment]
At the end this seems to be an issue with iotagent-node-lib, so the issue is transfered to that repo.
